### PR TITLE
build, multiarch: support splitting build logs for `--platform`

### PR DESCRIPTION
**Pull request by flouthoc**
Support splitting build logs for multi-arch builds ( `--platform` ) by
using `--logfile` and `--split-logfile-by-platform` which splits
generated log file in the convention `${file}_${platform_os}_${platform_arch}`

Closes: https://github.com/containers/buildah/issues/3960

#### Does this PR introduce a user-facing change?

```release-note

Add support for --split-logfile-by-platform which allows users to split logfile with build logs for each platform

```

### CODE REVIEW

File: imagebuildah/build.go [hunk 1]
```diff
@@ -253,6 +253,23 @@ func BuildDockerfiles(ctx context.Context, store storage.Store, options define.B
...
 		platformOptions.Args = argsCopy
 		builds.Go(func() error {
+			if platformOptions.LogFile != "" && platformOptions.LogSplitByPlatform {
+				logFile := platformOptions.LogFile + "_" + platformOptions.OS + "_" + platformOptions.Architecture
+				f, err := os.OpenFile(logFile, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
+				if err != nil {
+					return errors.Errorf("error opening logfile %q: %v", logFile, err)
+				}
+				defer f.Close()
+				logger = logrus.New()
```
Comment (nalind) [inline imagebuildah/build.go:263]: Isn't `logger` shared between all of these goroutines?

Comment (flouthoc) [reply imagebuildah/build.go:263]: In this case since logger for all the goroutine is writing to their own unique filepaths hence I thought its better to re-init since for this particular case sharing was not helping a lot since all the logs go to different files for each go-routine.

Comment (flouthoc) [reply imagebuildah/build.go]: Should be fixed now.

File: imagebuildah/build.go [hunk 2]
```diff
@@ -253,7 +253,25 @@ func BuildDockerfiles(ctx context.Context, store storage.Store, options define.B
 		}
 		platformOptions.Args = argsCopy
 		builds.Go(func() error {
-			thisID, thisRef, err := buildDockerfilesOnce(ctx, store, logger, logPrefix, platformOptions, paths, files)
+			loggerPerPlatform := logger
+			if platformOptions.LogFile != "" && platformOptions.LogSplitByPlatform {
+				logFile := platformOptions.LogFile + "_" + platformOptions.OS + "_" + platformOptions.Architecture
+				f, err := os.OpenFile(logFile, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
+				if err != nil {
+					return errors.Errorf("error opening logfile %q: %v", logFile, err)
```
Comment (nalind) [inline imagebuildah/build.go:261]: This should probably be `errors.Wrapf()`, or if we're using the standard library's `Errorf()`, it should use "%w" as the specifier for `err`.

File: pkg/cli/common.go [hunk 3]
```diff
@@ -211,6 +212,7 @@ func GetBudFlags(flags *BudResults) pflag.FlagSet {
 	fs.IntVar(&flags.Jobs, "jobs", 1, "how many stages to run in parallel")
 	fs.StringArrayVar(&flags.Label, "label", []string{}, "set metadata for an image (default [])")
 	fs.StringVar(&flags.Logfile, "logfile", "", "log to `file` instead of stdout/stderr")
+	fs.BoolVar(&flags.LogSplitByPlatform, "split-logfile-by-platform", false, "split logfile to different files for each platform")
```
Comment (rhatdan) [inline pkg/cli/common.go:215]: Could you shorten this to something like `--logsplit` I hate multi word options, and this might be the worst one ever.  Document the option in the help, do not need to have an explanation in the option name itself.  Also the rest of the world is not as good a typest as you.
